pkgs/build-support/setup-hooks/strip.sh

```diff
@@ -61,8 +61,15 @@ stripDirs() {
 
     if [ -n "${paths}" ]; then
         echo "stripping (with command $cmd and flags $stripFlags) in $paths"
+
+        # xargs exits with status code 123 if some but not all of the
+        # processes fail. We don't care if some of the files couldn't
+        # be stripped, so ignore specifically this code.
+
         # Do not strip lib/debug. This is a directory used by setup-hooks/separate-debug-info.sh.
-        find $paths -type f -a '!' -wholename "$prefix/lib/debug/*" -exec $cmd $stripFlags '{}' \; 2>/dev/null
+        find $paths -type f -a '!' -path "$prefix/lib/debug/*" -print0 |
+            xargs -r -0 -n1 -P "$NIX_BUILD_CORES" $cmd $stripFlags || [[ $? = 123 ]]
```

Comment:
Since the old code never exited non-zero, I don't think we want to introduce this behavior now.

Reply:
Do you have a concrete case where this would be a problem? I'd say the new semantics, while different, are more correct:
```
       xargs exits with the following status:

              0      if it succeeds
              123    if any invocation of the command exited with status 1-125
              124    if the command exited with status 255
              125    if the command is killed by a signal
              126    if the command cannot be run
              127    if the command is not found
              1      if some other error occurred.

       Exit codes greater than 128 are used by the shell to indicate that a program died due to a fatal signal.
```

particularly if the strip command is not found (124) or segfaults or is targeted by the OOM killer (125).

Reply:
Yep, I agree :)